nomad/structs/structs.go

```diff
@@ -5350,6 +5350,34 @@ func (t *Task) Validate(ephemeralDisk *EphemeralDisk, jobType string) error {
 		}
 	}
 
+	// Validation for Kind field which is used for Consul Connect integration
+	// TODO better wording for all error messages
+	if strings.HasPrefix(t.Kind, "connect:") {
```

Comment:
I wonder how much of a hassle it would be to use a special type with helper methods for Kind?

```go
type Kind string

func (k Kind) IsConnect() bool {
  const prefix = "connect:"
  return len(k) > len(prefix) && k[:len(prefix)] == "connect:"
}
```

That way our validation could just use the helpers:

```go
if t.Kind.IsConnect() { /* ... */ }
```

I don't *think* the custom type will confuse any marshaling, unmarshalling, parsing, etc.